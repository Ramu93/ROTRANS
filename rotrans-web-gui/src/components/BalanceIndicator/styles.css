.value-label {
  font-size: 14px;
  font-weight: 400;
  color: grey;
}

.value-text {
  margin-top: 12px;
  font-size: 20px;
  font-weight: 600;
  color: #6435c9;
}