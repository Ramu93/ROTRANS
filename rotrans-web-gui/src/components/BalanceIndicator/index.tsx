import React from "react";
import "./styles.css";

interface Props {
  balance: string;
}

const BalanceIndicator: React.FC<Props> = ({ balance }) => {
  return (
    <>
      <span className="value-label">Available Balance</span>
      <span data-testid="balanceIndicatorValueText" className="value-text">{balance}</span>
    </>
  );
};

export default BalanceIndicator;
