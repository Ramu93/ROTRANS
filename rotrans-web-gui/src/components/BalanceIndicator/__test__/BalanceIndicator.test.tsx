import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import BalanceIndicator from "..";

describe("BalanceIndicator component", () => {
  afterEach(cleanup);

  it("Rendered", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BalanceIndicator balance="123" />, div);
  });

  it("Rendered with props", () => {
    const { getByTestId } = render(<BalanceIndicator balance="123" />);
    expect(getByTestId("balanceIndicatorValueText")).toHaveTextContent("123");
  });

  it("Matches snapshot", () => {
    const tree = renderer.create(<BalanceIndicator balance="123" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
