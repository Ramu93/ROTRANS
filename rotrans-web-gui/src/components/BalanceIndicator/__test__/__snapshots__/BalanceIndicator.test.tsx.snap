// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceIndicator component Matches snapshot 1`] = `
Array [
  <span
    className="value-label"
  >
    Available Balance
  </span>,
  <span
    className="value-text"
    data-testid="balanceIndicatorValueText"
  >
    123
  </span>,
]
`;
